Include/object.h

```diff
@@ -261,6 +261,8 @@ _Py_ThreadId(void)
     __asm__ ("mrs %0, tpidrro_el0" : "=r" (tid));
 #elif defined(__aarch64__)
     __asm__ ("mrs %0, tpidr_el0" : "=r" (tid));
+#elif define(HAVE_BUILTIN_THREAD_POINTER)
```

Comment:
This line of the public C API which can be used by a C compiler different than the one used to build Python. I don't think that `if define(HAVE_BUILTIN_THREAD_POINTER)` check is reliable, according to what @colesbury wrote in the issue.

Reply:
I wrote "C compiler", but the C API can be parsed by parsers which are not C compiler, but written in any programming language. Moreover, the C API is used in C++.

Reply:
Yeah, I'm concerned that the compiler for CPython may be not exactly the same as the compiler for third-party extensions. For example, if we want to use `__builtin_thread_pointer()` for x86-64, it's available for GCC 11.1+, but an extension might be compiling with an older version of GCC that doesn't support it for that architecture.

Additionally, one thing to be careful of, is that macOS arm64 will successfully compile and run `__builtin_thread_pointer()`, but it returns the wrong value (CPU id).

PowerPC on GCC still requires a ["register variable"](https://gcc.gnu.org/onlinedocs/gcc/Global-Register-Variables.html) https://github.com/bminor/glibc/blob/master/sysdeps/powerpc/nptl/thread_pointer.h (or it can fallback to a portable thread-local implementation, but that would be slower.)